lib/lte_link_control/lte_lc.c

```diff
@@ -669,10 +667,6 @@ static int connect_lte(bool blocking)
 		goto exit;
 	}
 
-	if (blocking) {
-		k_sem_init(&link, 0, 1);
-	}
-
```

Comment:
Should this be replaced with `k_sem_reset()` instead? The `link` semaphore is given every time a `+CEREG` notification with status home or roaming is received. This works when `connect_lte()` is called for the first time, but I think this sequence does not work correctly:

1. `connect_lte()` called, LTE is connected and `connect_lte()` returns.
2. At least one `+CEREG` notification received with status `1` or `5`. The link semaphore is incremented.
3. Modem is put into offline mode.
4. `connect_lte()` is called again. The link semaphore is already `1` and the function returns immediately.

Reply:
Yes, you are correct. I will add a `k_sem_reset` before we change the functional mode, which is later than when we called `k_sem_init`. I think that is perhaps a bit more safe, plus we can do that unconditionally at that point.